doc/tools/make_rst.py

```diff
@@ -1219,13 +1225,16 @@ def make_rst_class(class_def: ClassDef, state: State, dry_run: bool, output_dir:
 
                     # Create annotation signature and anchor point.
 
+                    self_link = ""
                     if i == 0:
-                        f.write(f".. _class_{class_name}_annotation_{m.name}:\n\n")
+                        annotation_anchor = f"class_{class_name}_annotation_{m.name}"
+                        f.write(f".. _{annotation_anchor}:\n\n")
+                        self_link = f" :ref:`🔗<{annotation_anchor}>`"
 
                     f.write(".. rst-class:: classref-annotation\n\n")
 
                     _, signature = make_method_signature(class_def, m, "", state)
-                    f.write(f"{signature}\n\n")
+                    f.write(f"{signature}{self_link}\n\n")
```

Comment:
Any reason for some having a space before the self link and some not?

Reply:
If there is a self-link, there is always a space before it. In some cases I moved the space into the `self_link` variable, because there is a branch where `self_link = ""`, namely when no anchor was generated in the pre-existing code.